### ISSUE TITLE
docker libnetwork crash

### ISSUE THREAD

**Issue by rojingeorge** [docker/libnetwork#1422]
Hi Guys,
Docker daemon crashed, please have a check. Below where the things which I was doing when it crashed.

I run docker engine in cluster mode with below command.
```
docker daemon -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock --cluster-store=etcd://127.0.0.1:4001 --cluster-advertise=10.19.122.70:2375
```
Then I ran Calico with --libnetwork param with my HOSTIP, After that made docker network using calico driver
```
calicoctl node --libnetwork --ip=10.19.122.70
docker network create --driver=calico --subnet=10.10.10.0/16 RojinNetwork
```

After that started a container running over the new calico based network and did a traceroute and docker daemon crashed
```
docker run --net=RojinNetwork -v /usr/local/bin/:/root -itd --name=workerVolume-1 ubuntu:14.04 /bin/bash
```

**Crash Info**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x28 pc=0x8b3610]

goroutine 2915 [running]:
panic(0x1a52560, 0xc8200100e0)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/docker/libnetwork.(*resolver).ServeDNS(0xc8203e3170, 0x7fb5113f7d90, 0xc82183f3b0, 0xc8202d5440)
	/usr/src/docker/vendor/src/github.com/docker/libnetwork/resolver.go:360 +0x7b0
github.com/miekg/dns.(*Server).serve(0xc8202995f0, 0x7fb5113ebfe8, 0xc820fa92c0, 0x7fb51143d9a8, 0xc8203e3170, 0xc8203bdc00, 0x29, 0x200, 0xc8203709f8, 0xc820fa6620, ...)
	/usr/src/docker/vendor/src/github.com/miekg/dns/server.go:535 +0x7c8
created by github.com/miekg/dns.(*Server).serveUDP
	/usr/src/docker/vendor/src/github.com/miekg/dns/server.go:489 +0x3d5
```

**Comment by sanimej** [comment]
@rojingeorge What Docker version are you using ?

**Comment by rojingeorge** [comment]
@sanimej - Please find the info

```
root@rojin:~# docker version
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:00:36 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:00:36 2016
 OS/Arch:      linux/amd64
```

**Comment by aboch** [comment]
Fixed by #1459

**Comment by rojingeorge** [comment]
Thanks
